src/NuGet.Core/NuGet.Build.Tasks/GetRestoreNuGetAuditSuppressionsTask.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+
+namespace NuGet.Build.Tasks
+{
+    public class GetRestoreNuGetAuditSuppressionsTask : Microsoft.Build.Utilities.Task
+    {
+        /// <summary>
+        /// Full path to the msbuild project.
+        /// </summary>
+        [Required]
+        public string ProjectUniqueName { get; set; }
+
+        [Required]
+        public ITaskItem[] NuGetAuditSuppressions { get; set; }
+
+        /// <summary>
+        /// Target frameworks to apply this for. If empty this applies to all.
+        /// </summary>
+        public string TargetFrameworks { get; set; }
+
+        /// <summary>
+        /// Output items
+        /// </summary>
+        [Output]
+        public ITaskItem[] RestoreGraphItems { get; set; }
+
+        public override bool Execute()
+        {
+            var entries = new List<ITaskItem>();
+            var seenIds = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Duplicates certainly matter for PackageReference since you could have:
```xml
<PackageReference Include="A" Version="1.0.0" />
...
<PackageReference Include="A" Version="2.0.0" />
```
So we'd warn a user that they have specified A twice or more and they might not get the version they expect since MSBuild and NuGet have to decided which single one to use.  Would duplicate suppressions for advisories have this problem?  If not, I wouldn't bother de-duplicating them since it adds overhead.  But if a user could accidentally specify more than one and potentially not get the desired behavior, we'll have to warn them.

Reply:
I think what @zivkan is saying that by the time we get to [GetRestoreNuGetAuditSuppressionsTask.cs](https://github.com/NuGet/NuGet.Client/pull/5679/files/5e35b7e4c5577af50cf50e108043b8010a12f04d#diff-8bebcf007d7d681e507ab067017c52c1fbeffb425cff02a79d105271b9da9a97), things will be deduplicated (warnings and all) already, so we shouldn't need to check it twice..